File: Refatorar/Codigo Refatorar - Legado/Java/Atividade11.java
import static spark.Spark.*;
import java.sql.*;
import java.util.*;
import com.google.gson.Gson;

public class App {
    public static void main(String[] args) {
        port(3000);
        Gson g = new Gson();

        get("/prod", (req, res) -> {
            Connection c = DriverManager.getConnection("jdbc:mysql://localhost:3306/loja", "root", "");
            Statement s = c.createStatement();
            ResultSet rs = s.executeQuery("SELECT * FROM produtos");
            List<Map<String, Object>> l = new ArrayList<>();
            while (rs.next()) {
                Map<String, Object> m = new HashMap<>();
                m.put("id", rs.getInt("id"));
                m.put("n", rs.getString("nome"));
                m.put("p", rs.getDouble("preco"));
                l.add(m);
            }
            c.close();
            return g.toJson(l);
        });

        post("/prod", (req, res) -> {
            String b = req.body();
            Map<String, Object> m = g.fromJson(b, Map.class);
            String n = (String) m.get("n");
            Double p = (Double) m.get("p");
            Connection c = DriverManager.getConnection("jdbc:mysql://localhost:3306/loja", "root", "");
            PreparedStatement st = c.prepareStatement("INSERT INTO produtos (nome, preco) VALUES (?, ?)");
            st.setString(1, n);
            st.setDouble(2, p);
            st.executeUpdate();
            c.close();
            return "ok";
        });

        put("/prod/:id", (req, res) -> {
            int id = Integer.parseInt(req.params(":id"));
            String b = req.body();
            Map<String, Object> m = g.fromJson(b, Map.class);
            String n = (String) m.get("n");
            Double p = (Double) m.get("p");
            Connection c = DriverManager.getConnection("jdbc:mysql://localhost:3306/loja", "root", "");
            PreparedStatement st = c.prepareStatement("UPDATE produtos SET nome=?, preco=? WHERE id=?");
            st.setString(1, n);
            st.setDouble(2, p);
            st.setInt(3, id);
            st.executeUpdate();
            c.close();
            return "ok";
        });

        delete("/prod/:id", (req, res) -> {
            int id = Integer.parseInt(req.params(":id"));
            Connection c = DriverManager.getConnection("jdbc:mysql://localhost:3306/loja", "root", "");
            PreparedStatement st = c.prepareStatement("DELETE FROM produtos WHERE id=?");
            st.setInt(1, id);
            st.executeUpdate();
            c.close();
            return "ok";
        });

        for (int i = 0; i < 200; i++) {
            final int idx = i;
            get("/t" + i, (req, res) -> "t" + idx);
        }
    }
}

File: Refatorar/Codigo Refatorar - Legado/Typescript/Atividade11.ts
import express from 'express';
const app = express();
app.use(express.json());

let produtos: any[] = [];
let cont = 0;

function val(p: any): boolean {
    if (p.n && p.v) return true;
    return false;
}

function err() {
    return { e: 'inválido' };
}

app.get('/list', (req, res) => {
    let out = [];
    for (let i = 0; i < produtos.length; i++) {
        out.push({ id: i, nome: produtos[i].n, valor: produtos[i].v });
    }
    res.json(out);
});

app.post('/add', (req, res) => {
    let d = req.body;
    if (!val(d)) { res.json(err()); return; }
    produtos.push(d);
    res.json({ ok: true });
});

app.put('/up/:id', (req, res) => {
    let id = Number(req.params.id);
    let d = req.body;
    if (id >= produtos.length || id < 0) { res.json(err()); return; }
    produtos[id] = d;
    res.json({ ok: true });
});

app.delete('/rm/:id', (req, res) => {
    let id = Number(req.params.id);
    if (id >= produtos.length || id < 0) { res.json(err()); return; }
    produtos.splice(id, 1);
    res.json({ ok: true });
});

app.get('/ct', (req, res) => {
    res.json({ t: produtos.length });
});

for (let i = 0; i < 50; i++) {
    app.get(`/r${i}`, (req, res) => { res.send(`r${i}`); });
}

app.listen(3000);
